irohad/torii/impl/query_service.cpp

```diff
@@ -88,4 +90,71 @@ namespace torii {
+      grpc::ServerContext *context,
+      const iroha::protocol::BlocksQuery *request,
+      grpc::ServerWriter<iroha::protocol::BlockQueryResponse> *writer) {
+    log_->info("Fetching commits");
+    shared_model::proto::TransportBuilder<
+        shared_model::proto::BlocksQuery,
+        shared_model::validation::DefaultBlocksQueryValidator>()
+        .build(*request)
+        .match(
+            [this, context, request, writer](
+                const iroha::expected::Value<shared_model::proto::BlocksQuery>
+                    &query) {
+              rxcpp::composite_subscription sub;
+              query_processor_
+                  ->blocksQueryHandle(
+                      std::make_shared<shared_model::proto::BlocksQuery>(
+                          query.value))
+                  .as_blocking()
+                  .subscribe(
+                      sub,
+                      [this, context, &sub, request, writer](
+                          const std::shared_ptr<
+                              shared_model::interface::BlockQueryResponse>
+                              response) {
+                        if (context->IsCancelled()) {
+                          log_->info("Unsubscribed");
+                          sub.unsubscribe();
+                        } else {
+                          log_->info("{} receives committed block",
+                                     request->meta().creator_account_id());
+                          iroha::visit_in_place(
+                              response->get(),
+                              [writer](
+                                  const shared_model::interface::BlockResponse
+                                      &block_response) {
+                                auto proto_block_response = static_cast<
+                                    const shared_model::proto::BlockResponse &>(
+                                    block_response);
+                                writer->Write(
+                                    proto_block_response.getTransport());
+                              },
+                              [writer](const shared_model::interface::
+                                           BlockErrorResponse
+                                               &block_error_response) {
+                                auto proto_block_error_response =
+                                    static_cast<const shared_model::proto::
+                                                    BlockErrorResponse &>(
+                                        block_error_response);
+                                writer->WriteLast(
+                                    proto_block_error_response.getTransport(),
+                                    grpc::WriteOptions());
+                              });
+                        }
+                      });
+            },
+            [this, writer](const auto &error) {
+              log_->info("Stateless invalid: {}", error.error);
```

Comment:
It is either a warning, or an error

Reply:
Should iroha log a warning if it received stateless invalid query? I don't think so, this is legal behaviour for iroha

Reply:
Summon @luckychess